tests/test_random_number_generator.h

```diff
@@ -106,6 +105,81 @@ TEST_CASE("[RandomNumberGenerator] Restore from seed") {
 	CHECK_MESSAGE(s0_1_before == s0_1_after, msg);
 	CHECK_MESSAGE(s0_2_before == s0_2_after, msg);
 }
+
+TEST_CASE("[RandomNumberGenerator] Check rand range for expected functionality") {
+	Ref<RandomNumberGenerator> rng = memnew(RandomNumberGenerator);
+	rng->set_seed(0);
+	uint64_t prev_state = rng->get_state();
+	uint32_t seed_before = rng->get_seed();
+
+	String msg = "Randf range should give float between -100 and 100, base test";
+	float num_before = rng->randf_range(-100, 100);
+	CHECK_MESSAGE(-100 <= num_before, msg);
+	CHECK_MESSAGE(num_before <= 100, msg);
+	float num_after1 = rng->randf_range(-75, 75);
+	rng->randomize();
+
+	String msg2 = "Randf range should give float between -75 and 75.";
+	msg2 += " Shouldn't be affected by randomize.";
+	CHECK_MESSAGE(-75 <= num_after1, msg2);
+	CHECK_MESSAGE(num_after1 <= 75, msg2);
+	rng->set_state(prev_state);
+	int num_after2 = rng->randi_range(-50, 50);
+
+	String msg3 = "Randi range should give int between -50 and 50.";
+	msg3 += " Shouldn't be affected by set state.";
+	CHECK_MESSAGE(-50 <= num_after2, msg3);
+	CHECK_MESSAGE(num_after2 <= 50, msg3);
```

Comment:
I'd personally use `CHECK` over `CHECK_MESSAGE` because the underlying test framework is smart enough to compile a readable message out of simple assertions.

`INFO("Shouldn't be affected by set_state.")` could also be used for informational messages (they won't be printed unless the test case fails or `--test --success` is passed via command-line).